napari/utils/colormaps/colormap_utils.py

```diff
@@ -618,27 +618,26 @@ def ensure_colormap(colormap: ValidColormapArg) -> Colormap:
     """
     with AVAILABLE_COLORMAPS_LOCK:
         if isinstance(colormap, str):
-            # if the name starts with a '#' we assume it to be a hex color
-            # we then check if the hex color is one of r,g,b,c,m,y,w
-            # in which case we can reuse the associated colormap.
-            if colormap.startswith('#') and len(colormap) in [4, 5, 7, 9]:
-                name = _hexcolor_to_name(colormap)
-                if name.startswith('custom'):
-                    # generate a linear colormap from black to that color
-                    # and add it to the dictionary of available colormaps
-                    cmap = _colormap_from_colors(colormap, name)
-                    if cmap is not None:
-                        AVAILABLE_COLORMAPS[name] = cmap
-                    else:
-                        raise ValueError(
-                            trans._(
-                                "Unable to convert the hex color string into a colormap",
-                                deferred=True,
-                            )
+            #is a colormap with this name (default or custom) already available?
+            cmap = AVAILABLE_COLORMAPS.get(colormap, AVAILABLE_COLORMAPS.get(f"custom-{colormap.lower()}", None))
+            if cmap is None:
+                cmap = _colormap_from_colors(colormap, name=f"custom-{colormap.lower()}")
+                if cmap is None:
+                    raise ValueError(
+                        trans._(
+                            'Unrecognized Colormap String Value: {name}',
+                            deferred=True,
+                            name=colormap,
```

Comment:
here should be the call of `vispy_or_mpl_colormap` instead of raising an exception.

Reply:
Thanks, this makes a lot of sense and the generated error message is perfectly understandable:
```python
import napari.utils.colormaps.colormap_utils as cu
cmap = cu.ensure_colormap('some_name')
```
returns
```
KeyError: 'Colormap "some_name" not found in either vispy or matplotlib. Recognized colormaps are: "GrBu", "GrBu_d", "PiYG", "PuGr", "RdBu", "RdYeBuCy", "autumn", "blues", "cool", "coolwarm", "cubehelix", "diverging", "fire", "gist_earth", "gray", "gray_r", "grays", "greens", "hot", "hsl", "hsv", "husl", "ice", "inferno", "light_blues", "magma", "orange", "plasma", "reds", "single_hue", "spring", "summer", "turbo", "twilight", "twilight_shifted", "viridis", "winter"'
```
Any of the suggested names will work, for example 'twilight_shifted' will generate quite a complex colormap.

Let me know if anything else needs amending, and thank you for your reviews!

Cheers,
Egor